\section{Introduction}

Many processes that shape the distribution of biodiversity across a landscape
are expected to affect multiple species.
Examples include changes to the environment itself that can cause
co-distributed species to contract or expand their ranges or become isolated.
Furthermore, ecological interactions can also cause demographic changes in
multiple species;
for example, the expansion of one species may predict the expansion of its
symbionts and the contraction of competitors.
Such processes predict that population divergences or demographic changes will
be temporally clustered across multiple species.
As a result, statistical methods that infer such patterns from genetic data
allow phylogeographers to test hypotheses about such processes.
There has been a lot of work on methods that test for shared divergence times
among pairs of populations
\citep{Hickerson2006,Hickerson2007,Huang2011,Oaks2014dpp,Oaks2018ecoevolity}.
Recently, there has also been a lot of work on methods that infer temporally
clustered changes in population sizes among species
\citep{Chan2014,Xue2015,Xue2017,Gehara2017}.

Uses of ABC estimation of shared size changes \citep{Burbrink2016}

To date, no method has allowed the co-estimation of both shared divergences
and population-size changes.
Given the overlap in the processes that can potentially cause divergence and
demographic changes across multiple species, such a method would be useful for
testing hypotheses about community-scale processes that shape biodiversity
across landscapes.
Here, we introduce a general, full-likelihood Bayesian method that can estimate
shared times among an arbitrary mix of population divergences and population
size changes (\fig{}~\ref{fig:modelCartoon}).

Furthermore, there has been a lot of work on how well the divergence-time
methods work
\citep{Oaks2012,Hickerson2013,Oaks2014reply,Oaks2014dpp,Overcast2017,Oaks2018ecoevolity}.
However, work on how well we can estimate shared population-size changes has
been much more limited.
There are theoretical reasons to suspect that estimating shared demographic
changes is more difficult than divergence times.
The parameter of interest can become unidentifiable in at least three ways.
First, as the magnitude of a change in population becomes smaller,
it becomes more difficult to estimate.
Second, as the age of the size change gets older, we run out of
genetic coalescent events that allow us to learn when it happened.
Third, as the age of the size change approaches zero, we once again will fail
to sample the coalescent events needed to inform us of the change.
Thus, our second goal is to leverage our new method to assess how well we can
infer shared population size changes among species.

\ifembed{
\input{fig-model-cartoon.tex}
}{}


\section{Results \& Discussion}

\input{results.tex}

\section{Conclusions}

We are threading a needle.
There is a very narrow window within which we can reasonably estimate
time of a demographic change.
This window is determined by how deep in the past the change occurred
in coalescent units (i.e., relative to the current effective size
of the population).
Too recent and we lack enough coalescence events after the change, and too old
and we lack enough coalscence events before the change.
Even within this window, however, estimates of sharing of demographic
changes among populations are still quite poor.
This is in stark contrast to inferring sharing of divergences
among taxa.

Even when all of the information is used from allele count
data, we can't do it.
Caution should be taken from any inference of shared demographic changes.
Perhaps there is hope if we model alignments of contiguous sequences.
Longer loci can contain much more information about the lengths
of branches in the gene tree, which are critically informative
about the size of the population at any point in time.

\section{Methods}

\input{methods.tex}
