\section{Introduction}

\jcomment{Need more background, context, and citations}
A primary goal of ecology and evolutionary biology is to understand the
processes influencing the distribution, abundance, and diversification of
species.
Many biotic and abiotic factors that shape the distribution of biodiversity
across a landscape are expected to affect multiple species.
Abiotic mechanisms include changes to the environment that can cause
co-distributed species to contract or expand their ranges and/or become
fragmented.
Biotic factors include inter-specific ecological interactions such as the range
expansion of a species causing the expansion of its symbionts and the range
contraction and/or fragmentation of its competitors.
Such processes predict that evolutionary events, such as population divergences
or demographic changes, will be temporally clustered across multiple species.
As a result, statistical methods that infer such patterns from genetic data
allow ecologists and evolutionary biologists to test hypotheses about such
community-scale processes.

Recent research has developed methods to infer
patterns of temporally clustered (or ``shared'') evolutionary events,
including shared divergence times among pairs of populations
\citep{Hickerson2006,Hickerson2007,Huang2011,Oaks2014dpp,Oaks2018ecoevolity}
and shared demographic changes in effective population size across populations
\citep{Chan2014,Xue2015,Burbrink2016,Xue2017,Gehara2017}
from comparative genetic data.
% We use ``demography'' to refer to the effective size of populations.
To date, no method has allowed the joint inference of both shared divergences
and demographic changes.
Given overlap among processes that can potentially cause divergence and
demographic changes of populations across multiple species, such a method would
be useful for testing hypotheses about community-scale processes that shape
biodiversity across landscapes.
Here, we introduce a general, full-likelihood Bayesian method that can estimate
shared times among an arbitrary mix of population divergences and population
size changes (\fig{}~\ref{fig:modelCartoon}).
% We implemented this method in the software package \ecoevolity.

Whereas the theory and performance of methods that estimate shared divergence
times has been relatively well-investigated
\citep[e.g.,][]{Oaks2012,Hickerson2013,Oaks2014reply,Oaks2014dpp,Overcast2017,Oaks2018ecoevolity},
exploration into the estimation of shared demographic changes has been much
more limited.
There are theoretical reasons to suspect that estimating shared demographic
changes is more difficult than divergence times.
The parameter of interest (timing of a demographic change) is informed by
differing rates at which sampled copies of a gene ``find'' their common
ancestors going backward in time (coalescence) before and after the change in
population size, and this can become unidentifiable in three ways.
First, as the magnitude of a change in population becomes smaller,
it becomes more difficult to estimate.
Second, as the age of the demographic change gets older, fewer of the genetic
coalescent events occur prior to the change, resulting in less information
about the timing and magnitude of the change.
Third, as the age of the demographic change approaches zero, fewer coalescent
events occur after the change, again limiting information about the timing and
magnitude of the change.
Thus, we take advantage of our full-likelihood method to assess how well we can
infer shared demographic changes among populations when using all the
information in genomic data.
We apply our method to restriction-site-associated DNA sequence (RADseq) data
from five populations of three-spine stickleback
\citep[(\spp{Gasterosteus aculeatus});][]{Hohenlohe2010}
that were previously estimated to have co-expanded with an approximate Bayesian
computation (ABC) approach \citep{Xue2015}.

\ifembed{
\input{fig-model-cartoon.tex}
}{}

\section{Materials \& Methods}

\input{methods.tex}


\section{Results \& Discussion}

\input{results.tex}


\section{Conclusions}

% We are threading a needle.
There is a narrow temporal window within which we can reasonably estimate the
time of a demographic change.
This width of this window is determined by how deep in the past the change
occurred in coalescent units (i.e., relative to the current effective size of
the population).
Too recent and we lack enough coalescence events after the change, and too old
and we lack enough coalscence events before the change.
Even within this window, however, estimates of the number of demographic events
and the assignment of populations to those events are still quite poor.
This is in stark contrast to inferring shared divergence times among taxa.

Given the poor estimates of co-demographic changes, even when all the
information in the data are leveraged by a full-likelihood method, any
inference of shared demographic changes should be treated with caution.
This is especially true when only variable sites are analyzed.
Our estimates of the timing and sharing of demographic changes were
consistently poor across all the conditions we simulated when only variable
sites were used.
Furthermore, when the prior is more diffuse than the distributions that
generated the data (which is usually the case in empirical applications),
excluding constant sites makes the method strongly biased toward
inferring shared demographic events
(second row of \fig{}~\ref{fig:valsimsmodeldiffuse}).
These results help explain the stark contrast we see in our results from the
stickleback RADseq data when including versus excluding constant sites
(\fig{}~\ref{fig:sticklesummary}).

Perhaps modelling contiguous, linked sequence loci could help
improve estimates of shared demographic changes.
Longer loci can contain much more information about the lengths of branches in
the gene tree, which are critically informative about the size of the
population at any point in time.
With loci from across the genome, each with more information about the gene
tree they evolved along,
perhaps more information can be captured about changes in the rate of
coalescence through time.
\jcomment{cite and discuss ``Bayesian skyline''/``phylodynamic'' methods here}
\citep{Pybus2000,Strimmer2001,OpgenRhein2005,Drummond2005,Heled2008,Minin2008beast,Ho2011,Palacios2012,Palacios2012UAI,Stadler2013,Gill2013,Palacios2014,Lan2015,Karcher2016,Karcher2017,Faulkner2018,Karcher2019}
The skyline-plot method has also been extended to work with SNP frequency spectra \citep{LiuFu2015}

Another potential path forward could entail modeling recombination along large
regions of chromosomes.
By approximating the full coalescent process, many methods have been developed
to model recombination in a computationally feasible manner
\citep{McVean2005,Marjoram2006,Chen2009,Li2011,Sheehan2013,Schiffels2014,Rasmussen2014,Palacios2015}.
This could potentially glean additional information from genomic data about the
linkage patterns among sites.
