\section{Introduction}

\jcomment{Need more background, context, and citations}
Many processes that shape the distribution of biodiversity across a landscape
are expected to affect multiple species.
Examples include changes to the environment itself that can cause
co-distributed species to contract or expand their ranges or become isolated.
Furthermore, ecological interactions can also cause demographic changes in
multiple species;
for example, the expansion of one species may predict the expansion of its
symbionts and the contraction of its competitors.
Such processes predict that population divergences or demographic changes will
be temporally clustered across multiple species.
As a result, statistical methods that infer such patterns from genetic data
allow phylogeographers to test hypotheses about such community-scale processes.

There has been a lot of work on methods that infer shared divergence times
among pairs of populations
\citep{Hickerson2006,Hickerson2007,Huang2011,Oaks2014dpp,Oaks2018ecoevolity}.
and shared demographic changes across populations
\citep{Chan2014,Xue2015,Burbrink2016,Xue2017,Gehara2017}.
To date, no method has allowed the co-inference of both shared divergences
and population-size changes.
Given the overlap in the processes that can potentially cause isolation and
demographic changes of populations across multiple species, such a method would
be useful for testing hypotheses about community-scale processes that shape
biodiversity across landscapes.
Here, we introduce a general, full-likelihood Bayesian method that can estimate
shared times among an arbitrary mix of population divergences and population
size changes (\fig{}~\ref{fig:modelCartoon}).

Whereas the theory and performance of methods that estimate shared divergence
times has been well-explored
\citep[e.g.,][]{Oaks2012,Hickerson2013,Oaks2014reply,Oaks2014dpp,Overcast2017,Oaks2018ecoevolity},
exploration into the estimation of shared demographic changes has been much
more limited.
There are theoretical reasons to suspect that estimating shared demographic
changes is more difficult than divergence times.
The parameter of interest, the timing of a demographic change, is informed by
differing rates of genetic coalescence before and after the change, and can
become unidentifiable in three ways.
First, as the magnitude of a change in population becomes smaller,
it becomes more difficult to estimate.
Second, as the age of the demographic change gets older, fewer of the genetic
coalescent events occur prior to the change, resulting in less information
about the timing and magnitude of the change.
Third, as the age of the demographic change approaches zero, fewer coalescent
events occur after the change, again limiting information about the timing and
magnitude of the change.
Thus, we take advantage of our full-likelihood method to assess how well we can
infer shared demographic changes among populations when using all the
information in genomic data.
We apply our method to restriction-site-associated DNA sequence (RADseq) data
from five populations of three-spine stickleback that were previously estimated
to have co-expanded with an approximate Bayesian computation (ABC) approach.

We find that estimating the timing and sharing of demographic changes is much
more challenging than divergence times.
Even under the most favorable simulation conditions, the ability to infer
shared demographic events is quite limited.
Our results also suggest that previous estimates of co-expansion among
stickleback populations are likely caused by lack of information in
single-nucleotide polymorphism (SNP) data that ignores invariant sites.

\ifembed{
\input{fig-model-cartoon.tex}
}{}


\section{Results \& Discussion}

\input{results.tex}

\section{Conclusions}

% We are threading a needle.
There is a very narrow window within which we can reasonably estimate the time
of a demographic change.
This window is determined by how deep in the past the change occurred
in coalescent units (i.e., relative to the current effective size
of the population).
Too recent and we lack enough coalescence events after the change, and too old
and we lack enough coalscence events before the change.
Even within this window, however, estimates of the number of demographic events
and the assignment of populations to those events are still quite poor.
This is in stark contrast to inferring shared divergence times among taxa.

Given the poor estimates of co-demographic changes, even when all
the information in the data are leveraged by a full-likelihood method, any
inference of shared demographic changes should be treated with caution.
This is especially true when only only variable sites are analyzed.
Our estimates of the timing and sharing of demographic changes were
consistently poor across all the conditions we simulated when only variable
sites were used.
Furthermore, when the prior is more diffuse than the distributions that
generated the data (which is usually the case in empirical applications),
excluding constant sites makes the method strongly biased toward
inferring shared demographic events
(second row of \fig{}~\ref{fig:valsimsmodeldiffuse}).
These results help explain the stark contrast we see in our results from the
stickleback RADseq data when including versus excluding constant sites
(\fig{}~\ref{fig:sticklesummary}).

Perhaps modelling contiguous, linked sequence loci could help
improve estimates of shared demographic changes.
Longer loci can contain much more information about the lengths of branches in
the gene tree, which are critically informative about the size of the
population at any point in time.
With loci from across the genome, each with more information about the gene
tree they evolved along,
perhaps more information can be captured about changes in the rate of
coalescence through time.
\jcomment{cite and discuss ``Bayesian skyline''/``phylodynamic'' methods here}


\section{Methods}

\input{methods.tex}
