\section{Introduction}

A primary goal of ecology and evolutionary biology is to understand the
processes influencing the distribution, abundance, and diversification of
species.
Many biotic and abiotic factors that shape the distribution of biodiversity
across a landscape are expected to affect multiple species.
Abiotic mechanisms include changes to the environment that can cause
co-distributed species to contract or expand their ranges and/or become
fragmented \citep{Wegener1966,Avise1987,Knowles2002}.
Biotic factors include inter-specific ecological interactions such as the range
expansion of a species causing the expansion of its symbionts and the range
contraction and/or fragmentation of its competitors
\citep{Lotka1920,Volterra1926}.
Such processes predict that evolutionary events, such as population divergences
or demographic changes, will be temporally clustered across multiple species.
As a result, statistical methods that infer such patterns from genetic data
allow ecologists and evolutionary biologists to test hypotheses about such
processes operating at or above the scale of communities of species.

Recent research has developed methods to infer
patterns of temporally clustered (or ``shared'') evolutionary events,
including shared divergence times among pairs of populations
\citep{Hickerson2006,Hickerson2007,Huang2011,Oaks2014dpp,Oaks2018ecoevolity}
and shared demographic changes in effective population size across populations
\citep{Chan2014,Xue2015,Burbrink2016,Xue2017,Gehara2017}
from comparative genetic data.
% We use ``demography'' to refer to the effective size of populations.
To date, no method has allowed the joint inference of both shared divergences
and population-size changes.
Given the overlap among processes that can potentially cause divergence and
demographic changes of populations across multiple species, such a method would
be useful for testing hypotheses about community-scale processes that shape
biodiversity across landscapes.
Here, we introduce a general, full-likelihood Bayesian method that can estimate
shared times among an arbitrary mix of population divergences and population
size changes (\fig{}~\ref{fig:modelCartoon}).
% We implemented this method in the software package \ecoevolity.

Whereas the theory and performance of methods that estimate shared divergence
times has been relatively well-investigated
\citep[e.g.,][]{Oaks2012,Hickerson2013,Oaks2014reply,Oaks2014dpp,Overcast2017,Oaks2018ecoevolity},
exploration into the estimation of shared changes in population size has been
much more limited.
There are theoretical reasons to suspect that estimating shared changes in
effective population size is more difficult than divergence times.
The parameter of interest (timing of a demographic change) is informed by
differing rates at which sampled copies of a gene ``find'' their common
ancestors (coalesce) going backward in time before and after the change in
population size, and this can become unidentifiable in three ways.
First, as the magnitude of a change in population becomes smaller,
it becomes more difficult to estimate.
Second, as the age of the demographic change gets older, fewer of the genetic
coalescent events occur prior to the change, resulting in less information
about the timing and magnitude of the change.
Third, information also decreases as the age of the demographic change
approaches zero, because fewer coalescent events occur after the change.
To explore these potential problems, we take advantage of our full-likelihood
method to assess how well we can infer shared demographic changes among
populations when using all the information in genomic data.
We apply our method to restriction-site-associated DNA sequence (RADseq) data
from five populations of three-spine stickleback
\citep[\spp{Gasterosteus aculeatus};][]{Hohenlohe2010}
that were previously estimated to have co-expanded with an approximate Bayesian
computation (ABC) approach \citep{Xue2015}.

\ifembed{
\input{fig-model-cartoon.tex}
}{}

\input{model.tex}

\section{Materials \& Methods}

\input{methods.tex}


\section{Results \& Discussion}

\input{results.tex}


\section{Conclusions}

% We are threading a needle.
There is a narrow temporal window within which we can reasonably estimate the
time of a demographic change.
The width of this window is determined by how deep in the past the change
occurred relative to the effective size of the population (i.e., in coalescent
units).
If too old or recent, there are too few coalescence events before or after the
demographic change, respectively, to provide information about the effective
size of the population.
When we are careful to simulate data within this window, and the change in
population size is large enough, we can estimate the time of the demographic
changes reasonably well
(e.g., see the top row of \fig{}~\ref{fig:valsimsetimesopt}).
However, even under these condtions, the ability to correctly infer the number
of demographic events, and the assignment of populations to those events is
quite limited
(\fig{}~\ref{fig:valsimsmodelopt}).
When only variable characters are analyzed (i.e., SNPs), estimates of
the timing and sharing of demographic changes are consistently bad; we see this
across all the conditions we simulated.
Most alarmingly, when the priors are more diffuse than the distributions that
generated the data, as will be true in most empirical applications, there is a
strong bias toward estimating too few demographic events
(i.e., over-clustering comparisons to demographic events;
Row 2 of \fig{}~\ref{fig:valsimsmodeldiffuse}),
especially when only variable characters are analyzed.
These results help explain the stark contrast we see in our results from the
stickleback RADseq data when including versus excluding constant sites
(\fig{}~\ref{fig:sticklesummary}).
These findings are in sharp contrast to estimating shared \emph{divergence}
times, which is much more accurate, precise, and robust to prior assumptions
\citep[\figs \labelcref{fig:valsimsmodeldiffuse,fig:valsimsetimesdiffuse,fig:mixsims}][]{Oaks2018ecoevolity,Oaks2018paic}.

Given the poor estimates of co-demographic changes, even when all the
information in the data are leveraged by a full-likelihood method, any
inference of shared demographic changes should be treated with caution.
However, there are potential ways that estimates of shared demographic
events could be improved.
For example, modelling loci of contiguous, linked sites could help
extract more information about past demographic changes.
Longer loci can contain much more information about the lengths of branches in
the gene tree, which are critically informative about the size of the
population through time.
This is evidenced by the extensive literature on powerful 
``skyline plot'' and ``phylodynamic'' methods
\citep{Pybus2000,Strimmer2001,OpgenRhein2005,Drummond2005,Heled2008,Minin2008beast,Ho2011,Palacios2012,Palacios2012UAI,Stadler2013,Gill2013,Palacios2014,Lan2015,Karcher2016,Karcher2017,Faulkner2018,Karcher2019}.
With loci from across the genome, each with more information about the gene
tree they evolved along,
perhaps more information can be captured about temporally clustered changes in
the rate of coalescence across populations.
Another potential source of information could be captured by modelling
recombination along large regions of chromosomes.
By approximating the full coalescent process, many methods have been developed
to model recombination in a computationally feasible manner
\citep{McVean2005,Marjoram2006,Chen2009,Li2011,Sheehan2013,Schiffels2014,Rasmussen2014,Palacios2015}.
This could potentially leverage additional information from genomic data about the
linkage patterns among sites along chromosomes.
Both of the these avenues are worth pursuing given the myriad historical
processes that predict patterns of temporally clustered changes in population
sizes across populations.
