\section{Introduction}

Many processes that shape the distribution of biodiversity across a landscape
are expected to affect multiple species.
Examples include changes to the environment itself that can cause
co-distributed species to contract or expand their ranges or become isolated.
Furthermore, ecological interactions can also cause demographic changes in
multiple species;
for example, the expansion of one species may predict the expansion of its
symbionts and the contraction of competitors.
Such processes predict that population divergences or demographic changes will
be temporally clustered across multiple species.
As a result, statistical methods that infer such patterns from genetic data
allow phylogeographers to test hypotheses about such processes.
There has been a lot of work on methods that test for shared divergence times
among pairs of populations
\citep{Hickerson2006,Hickerson2007,Huang2011,Oaks2014dpp,Oaks2018ecoevolity}.
Recently, there has also been a lot of work on methods that infer temporally
clustered changes in population sizes among species
\citep{Chan2014,Xue2015,Xue2017,Gehara2017}.

Uses of ABC estimation of shared size changes \citep{Burbrink2016}

To date, no method has allowed the co-estimation of both shared divergences
and population-size changes.
Given the overlap in the processes that can potentially cause divergence and
demographic changes across multiple species, such a method would be useful for
testing hypotheses about community-scale processes that shape biodiversity
across landscapes.
Here, we introduce a general, full-likelihood Bayesian method that can estimate
shared times among an arbitrary mix of population divergences and population
size changes (\fig{}~\ref{fig:modelCartoon}).

Furthermore, there has been a lot of work on how well the divergence-time
methods work
\citep{Oaks2012,Hickerson2013,Oaks2014reply,Oaks2014dpp,Overcast2017,Oaks2018ecoevolity}.
However, work on how well we can estimate shared population-size changes has
been much more limited.
There are theoretical reasons to suspect that estimating shared demographic
changes is more difficult than divergence times.
The parameter of interest can become unidentifiable in at least three ways.
First, as the magnitude of a change in population becomes smaller,
it becomes more difficult to estimate.
Second, as the age of the size change gets older, we run out of
genetic coalescent events that allow us to learn when it happened.
Third, as the age of the size change approaches zero, we once again will fail
to sample the coalescent events needed to inform us of the change.
Thus, our second goal is to leverage our new method to assess how well we can
infer shared population size changes among species.

\ifembed{
\input{fig-model-cartoon.tex}
}{}


\section{Methods}

\subsection{The data}
As described by \citet{Oaks2018ecoevolity},
we assume we have orthologous, biallelic genetic characters collected from
taxa we wish to compare.
By biallelic, we mean that each character has at most two states,
which we refer to as ``red'' and ``green'' following \citet{Bryant2012}.
For each taxon, we either have these data from one or more individuals from a
single population, in which case we infer the timing and extent of a population
size change, or one or more individuals from two populations or species,
in which case we infer the time when they diverged (\fig{}~\ref{fig:modelCartoon}).

For each population and for each character we genotype \allelecount copies
of the locus, \redallelecount of which are copies of the red allele and the
remaining $\allelecount - \redallelecount$ are copies of the green allele.
Thus, for each population of a pair, and for each locus, we have a count of the
total sampled gene copies and how many of those are the red allele.

\ifembed{
\input{../../tables/notation.tex}
}{}

Following the notation of \citet{Oaks2018ecoevolity}
we will use \leafallelecounts and \leafredallelecounts to denote allele counts
for a locus from either one population if we are modeling population-size
change or both populations of a pair if we are modeling a divergence; i.e., 
$\leafallelecounts, \leafredallelecounts = (\allelecount, \redallelecount)$
or
$\leafallelecounts, \leafredallelecounts = (\allelecount[1],
\redallelecount[1]), (\allelecount[2], \redallelecount[2])$
(\fig{}~\ref{fig:modelCartoon}
and
Table~\ref{table:notation}).
For convenience will use \comparisondata[i] to denote these allele counts
across all the loci from taxon $i$, which can be a single population
or a pair of populations.
Finally, we use \alldata to represent the data across all the taxa for which we
wish to compare times of either divergence or population-size change.
Note, because the population of each compared taxon is modeled separately
(\fig{}~\labelcref{fig:modelCartoon,fig:dag}),
characters do not have to be orthologous across taxa, only within them.


\subsubsection{The evolution of markers}

We assume each character evolved along a gene tree (\genetree)
according to a finite-sites, continuous-time Markov chain (CTMC) model.
We assume the gene tree of each character is independent of the others,
conditional on the population history (i.e., the characters are effectively
unlinked).
As the marker evolves along the gene tree, forward in time, there is an
instantaneous relative rate \rgmurate of mutating from the red state to the
green state, and a corresponding relative rate \grmurate of mutation from
green to red \citep{Bryant2012,Oaks2018ecoevolity}.

As the marker evolves along the gene tree, forward in time, there is an
instantaneous relative rate \rgmurate of mutating from the red state to the
green state, and a corresponding relative rate \grmurate of mutation from green
to red.
The stationary frequency of the green state is then
$\gfreq = \rgmurate / (\rgmurate + \grmurate)$.
We will use \murate to denote the overall rate of mutation.
If $\murate = 1$, time is measured in
units of expected substitutions per site.
Alternatively, if a mutation rate per site per unit time is given, then time is
absolute.

\subsubsection{The evolution of gene trees}

We assume that the gene tree of each locus evolved within a simple
``species'' tree with one ancestral root population, which either
left one or two descendant branches with different effective population sizes
at time \comparisondivtime
(\fig{}~\ref{fig:modelCartoon}).
We will use
\comparisonpopsizes{}
to denote all the effective population sizes of a species tree;
\epopsize[\rootpopindex] and 
\epopsize[\descendantpopindex{1}] when modeling a population-size change, and
(\epopsize[\rootpopindex],
\epopsize[\descendantpopindex{1}],
and \epopsize[\descendantpopindex{2}] when modeling a divergence.
Following \citet{Oaks2018ecoevolity}, we use
\sptree{}
as shorthand for the species tree, which comprises the population sizes and
divergence time of a pair
(\comparisonpopsizes{} and \comparisondivtime{}).


\subsubsection{The likelihood}

As in \citet{Oaks2018ecoevolity},
we use the work of \citet{Bryant2012}
to analytically integrate over all possible gene trees and
character substitution histories to compute the likelihood
of the species tree directly from 
a biallelic character pattern under a coalescent model;
$\pr(\leafallelecounts, \leafredallelecounts \given \sptree, \murate, \gfreq)$.
The only difference that is necessary is for population-size-change models that
have a species tree with only one descendant population.
Equation 19 of \citet{Bryant2012} shows how to obtain the partial likelihoods
at the bottom of an ancestral branch from the partial likelihoods at the top of
its two descendant branches.
When there is only one descendant branch, this is simplified, and the partial
likelihoods at the bottom of the ancestral branch are equal to the partial
likelihoods at the top of its sole descendant branch.
Other than this small change, the probability of a biallelic character pattern
given the species tree, mutation rate, and equilibrium state frequencies
($\pr(\leafallelecounts, \leafredallelecounts \given \sptree, \murate, \gfreq)$)
is calculated the same as in \citet{Bryant2012} and \citet{Oaks2018ecoevolity}.


\begin{linenomath}
For a given taxon, we can calculate the probability of all \nloci{} loci
from which we have data given the species tree and other parameters by
assuming independence among loci (conditional on the species tree) and
taking the product over them,
\begin{equation}
    \pr(\comparisondata \given \sptree, \murate, \gfreq)
    =
    \prod_{i=1}^{\nloci}
    \pr(\leafallelecounts[i], \leafredallelecounts[i] \given \sptree, \murate, \gfreq).
    \label{eq:comparisonlikelihood}
\end{equation}
We assume we biallelic data from \ncomparisons{} taxa, which can be an
arbitrary mix of
(1) two populations or species for which \comparisondivtime represents
the time they diverged, or
(2) one population for which \comparisondivtime represents the time
of a change in population size.
The likelihood across all \ncomparisons{} taxa is simply the product of the
likelihood of each taxon,
\begin{equation}
    \pr(
    \alldata
    \given
    \sptrees,
    \murates,
    \gfreqs)
    =
    \prod_{i=1}^{\ncomparisons}
    \pr(\comparisondata[i] \given \sptree[i], \murate[i], \gfreq[i]),
    \label{eq:collectionlikelihood}
\end{equation}
where
$\alldata = \comparisondata[1], \comparisondata[2], \ldots, \comparisondata[\ncomparisons]$,
$\sptrees = \sptree[1], \sptree[2], \ldots, \sptree[\ncomparisons]$,
$\murates = \murate[1], \murate[2], \ldots, \murate[\ncomparisons]$,
and
$\gfreqs = \gfreq[1], \gfreq[2], \ldots, \gfreq[\ncomparisons]$.
\end{linenomath}
As described in \citet{Oaks2018ecoevolity}, we condition the likelihood
for taxa for which we only sample variable characters.


\subsection{Bayesian inference}

\subsubsection{Priors}

\paragraph{Prior on the concentration parameter}

\paragraph{Prior on the event times}

\paragraph{Prior on mutation rates}

\paragraph{Prior on the equilibrium-state frequency}

\subsection{Software implementation}

\subsection{Analyses of simulated data}

\subsubsection{Assessing ability to estimate timing and sharing of demographic changes}

Settings used for all simulations include:
\begin{itemize}
    \item 3 pairs
    \item 500k characters
    \item 20 gene copies per population (10 diploid individuals)
    \item Concentration of DP 1.414216 (mean nevents of 2)
\end{itemize}

Settings used for all analyses include:
\begin{itemize}
    \item MCMC chain length of 75,000 sampled every 50 generations.
\end{itemize}

Initial pass. motivation: distribute event times to span values that are easy and hard.

Event times (in units of expected subsitutions per site) were exponentially
distributed with a mean of 0.01.
Considering the expected population size was 0.002, this puts our expectaction
for the event times in units of $4N_e$ generations at 1.25.
Thus, we expected to get a mix of demographic-change time that occurred more recently
than the gene trees coalesced and thus would be easy to estimate,
and those that occurred when few or no gene lineages were left to coalesce, and thus would
be difficult to estimate.

Results were quite poor across all conditions, but slightly better for the most dramatic pop expansions.
In an effort to find conditions where we could estimate demographic change times reasonble, we next simulated conditions where strong population expansions (relative root size gamma(10, mean = 0.25) and gamma(10, mean = 0.5)) occurred very recently (exponential with mean 0.001).

The improved performance was quite modest, and we ran into numerical issues when the
population expansion so recent that it was difficult to identify.
In these cases, the data were well explained by no expansion, which could be achieved in two ways: an expansion time of zero and an ancestral population size that matched the true ancestral population size, or an old expansion and a ldescendant population size that matched the true ancestral population size. The latter explained the data equally well when the divergence time was older than gene tree coalescences. This lead to MCMC chains converging to these different regions of parameter space.

This lead us to try tighter distribution on times with an offest to avoid
near-zero values, as well as offsest on the size of descendant populations and
ancestral relative sizes to avoid near-zero values (which cause rapid
coalescence, and thus no signal). The prior conditions we simulted under are:


\subsubsection{But what about under realistic prior information (diffuseprior)}

It was a lot of work to find conditions where we could reasonably estimate
the timing and number of demographic change events.
These condtions include quite informative distributions on parameters,
especially the relative size of the ancestral population.
We used the same distribution as the prior, and thus the informative priors
could be why the behavior is better (as opposed to the data containing a strong
signal of what happened).
To determine this, we simulated data under the best conditions,
but then analyzed these data under diffuse priors to see how
well we can expect to estimate.
This is important because in real world applications we usually know very
little about the timing and magnitude (and direction) of size changes.

Sim disributions:

Event time ~ gamma(shape=4.0, scale=0.000475, offset=0.0001); mean 0.002

relative root size ~ gamma(shape=5.0, scale = 0.04, offset = 0.05); mean 0.25

relative root size ~ gamma(shape=5.0, scale = 0.04, offset = 3.8); mean 4.0

descendant size ~ gamma(4.0, scale=0.0005, offset = 0.0001); mean 0.0021

Priors:

Event time ~ exponential(mean = 0.005)

relative root size ~ Exponential(mean = 2.0)

descendant size ~ gamma(2.0, scale 0.001); mean 0.002



\subsection{Simulating linked sites}
Should we analyze all the sites or just unlinked SNPs?


\subsection{Simulating a mix of divergence and pop-size-change comparisons}
What if we have a mix of pairs?

\subsection{Empirical application}
Stickleback data.


\section{Results}

\subsection{Analyses of simulated data}


\subsection{Empirical application}


\section{Discussion}
