\section{Introduction}

A primary goal of ecology and evolutionary biology is to understand the
processes influencing the distribution, abundance, and diversification of
species.
Many biotic and abiotic factors that shape the distribution of biodiversity
across a landscape are expected to affect multiple species.
Abiotic mechanisms include changes to the environment that can cause
co-distributed species to contract or expand their ranges and/or become
fragmented \citep{Hairston1960,Wegener1966,Avise1987,Knowles2002}.
Biotic factors include inter-specific ecological interactions such as the
population expansion of a species causing the expansion of its symbionts and
the population contraction and/or fragmentation of its competitors
\citep{Lotka1920,Volterra1926,Hairston1960,Hardin1960,Begon1996,Lunau2004}.
Such processes predict that evolutionary events, such as population divergences
or demographic changes, will be temporally clustered across multiple species.
As a result, statistical methods that infer such patterns from genetic data
allow ecologists and evolutionary biologists to test hypotheses about such
processes operating at or above the scale of communities of species.

Recently, researchers have developed methods to infer
patterns of temporally clustered (or ``shared'') evolutionary events,
including shared divergence times among pairs of populations
\citep{Hickerson2006,Hickerson2007,Huang2011,Oaks2014dpp,Oaks2018ecoevolity}
and shared demographic changes in effective population size across populations
\citep{Chan2014,Xue2015,Burbrink2016,Prates2016,Xue2017,Gehara2017}
from comparative genetic data.
% We use ``demography'' to refer to the effective size of populations.
To date, no method has allowed the joint inference of both shared divergences
and population-size changes.
Given the overlap among processes that can potentially cause divergence and
demographic changes of populations across multiple species, such a method would
be useful for testing hypotheses about community-scale processes that shape
biodiversity across landscapes.
Here, we introduce a general, full-likelihood Bayesian method that can estimate
shared times among an arbitrary mix of population divergences and population
size changes (\fig{}~\ref{fig:modelCartoon}).
% We implemented this method in the software package \ecoevolity.

Whereas the theory and performance of methods that estimate shared divergence
times has been relatively well-investigated
\citep[e.g.,][]{Oaks2012,Hickerson2013,Oaks2014reply,Oaks2014dpp,Overcast2017,Oaks2018ecoevolity},
exploration into the estimation of shared changes in population size has been
much more limited.
There are theoretical reasons to suspect that estimating shared changes in
effective population size is more difficult than divergence times
\citep{Myers2008}.
The parameter of interest (timing of a demographic change) is informed by
differing rates at which sampled copies of a locus ``find'' their common
ancestors (coalesce) going backward in time before and after the change in
population size, and this can become unidentifiable in three ways.
First, as the magnitude of the change in population size becomes smaller, it
becomes more difficult to identify, because the rates of coalescence before and
after the change become more similar.
Second, as the age of the demographic change increases, fewer of the genetic
coalescent events occur prior to the change, resulting in less information
about the effective size of the population prior to the change, and thus less
information about the magnitude and timing of the population-size change
itself.
\jedit{}{%
Third, information also decreases as the age of the demographic change
approaches zero, because fewer coalescent events occur after the change.}

\jedit{}{%
To explore these potential problems, we take advantage of our full-likelihood
method to assess how well we can infer shared demographic changes among
populations when using all the information in genomic data.}
We apply our method to restriction-site-associated DNA sequence (RADseq) data
from five populations of three-spine stickleback
\citep[\spp{Gasterosteus aculeatus};][]{Hohenlohe2010}
that were previously estimated to have co-expanded with an approximate Bayesian
computation (ABC) approach \citep{Xue2015}.
In stark contrast to shared divergence times, our results show that estimates
of shared changes in population size are quite poor across a broad range of 
simulation conditions.
We also find strikingly different estimates of the demographic histories of the
stickleback populations depending on whether we include invariant sites in
analyses.
This alarming result makes sense in light of the inference pathologies
exhibited by our analyses of simulated data,
where limited information in the data coupled with limited prior knowledge
about parameters leads to spurious support for shared demographic changes
across populations.

\ifembed{
\input{fig-model-cartoon.tex}
}{}

\input{model.tex}

\section{Materials \& Methods}

\input{methods.tex}


\section{Results \& Discussion}

\input{results.tex}

\subsection{Biological realism of our model of shared demographic changes}
The model of shared population-size changes we present above, and used in
previous research \citep{Chan2014,Xue2015,Gehara2017,Xue2015}, is quite
unrealistic in number of ways.
Modeling the demographic history of a population with a single,
instantaneous change in population size does not reflect the continuous and
complex demographic changes most populations of organisms experience through
time.
However, this simple model is correct in our simulated data, and yet our method
struggles to accurately infer the timing and sharing of these single, dramatic,
instantaneous changes in effective population size.
Incorporating more demographic realism into the model will introduce more
variation and thus make the inference problem even more difficult.
Thus, until inference of shared events under overly simplistic demographic
models can be improved, it does not seem wise to introduce more complexity.

Also, we expect most processes that cause shared divergences and/or demographic
changes across species will affect multiple species with some amount of
temporal variation.
Thus, our model of simultaneous evolutionary events that affect multiple
species at the same instant is not biologically plausible.
If this lack of realism is problematic, it should cause the method to
overestimate the number of events by misidentifying the temporal variation among
species affected by the same process as being the result of multiple events.
However, what we see here
(e.g., \fig{}~\ref{fig:valsimsmodeldiffuseonly})
and what has been shown previously
\citep{Oaks2012,Oaks2014reply,Oaks2014dpp,Oaks2018ecoevolity,Oaks2018paic}
is the opposite;
even when we model shared events as simultaneous, methods tend to underestimate
(and almost never overestimate) the number of events.
We do see overestimates when there is little information in the data
and the posterior largely reflects the prior 
(e.g., bottom two rows of \fig{}~\ref{fig:valsimsmodelopt}).
However, this is only true when the prior distributions match the true
underlying distributions that generated the data, and these overestimates would
be easy to identify in practice by testing for prior sensitivity and noticing
that the posterior probabilities of event models are similar to the prior
probabilities (i.e., small Bayes factors).
Furthermore, \citet{Oaks2018paic} showed that even with millions of bases of
genomic data from pairs of gecko populations, \ecoevolity was only able to
detect differences in divergence times between comparisons greater than several
thousand years.
Thus, it seems unlikely that over-estimating the number of events among taxa
(i.e., estimating temporal independence of comparisons that shared the same
historical process) is a real problem for these types of inferences.

Previous researchers
\citep{Overcast2017,Gehara2017,Xue2017}
have attempted to introduce realism into these comparative models by allowing
temporal variation among species affected by the same event, by assuming that
processes of diversification and demographic change are temporally
overdispersed.
However, allowing temporal variation within events will only increase the
tendency of these methods to underestimate the number of events (i.e., the
within-event temporal variation makes it ``easier'' to assign comparisons to
the same event).
More fundamentally, it seems odd to assume \emph{a priori} that processes that
cause shared evolutionary responses would be somehow conveniently staggered
over evolutionary timescales (overdispersed);
this seems like something we would want to estimate from the data.

\subsection{Comparison to previous models of shared demographic changes}

Our method is the first that we know of that is generalized to infer an
arbitrary mix of shared times of divergence and changes in population size.
However, if we focus only on changes in population size, the models underlying
the ABC methods of
\citet{Chan2014}, \citet{Xue2015}, and \citet{Gehara2017}
share many similarities with the model we introduced above.
These models, like ours, allow the effective population sizes before and after
the time of the demographic change to vary ($2\ncomparisons$ free parameters),
however, they assume all populations experienced an expansion.
The models of \citet{Chan2014}, \citet{Xue2015}, and \citet{Gehara2017}
also assume there was at most one shared demographic event;
each comparison can either be assigned to this event or have an independent
time of demographic change.
\citet{Xue2017} relaxed these constraints by allowing population contractions
and expansions and allowing any number of demographic events and assignments of
populations to those events, like we do here.
All previous approaches, like ours, model variation in gene trees using the
coalescent.
They also assume an infinite-sites model of character evolution along gene
trees, whereas our approach uses a finite-sites model.
\citet{Gehara2017} and \citet{Xue2017}
allow the investigator to assume that the processes that cause demographic
changes are temporally overdispersed (i.e., separated in time by ``buffers'').
We do not explore this temporal staggering of events here
because this is a pattern we would like to infer from data rather than impose
\emph{a priori}.
Furthermore, creating temporal ``buffers'' around events will exacerbate the
tendency to over-cluster comparisons (i.e., underestimate the number of
events).

The biggest difference between previous approaches and ours is how the data are
used.
\citet{Chan2014} and \citet{Gehara2017}
reduce aligned sequences into a set of population genetic summary statistics.
\citet{Xue2015} and \citet{Xue2017}
reduce SNPs into a site-frequency spectrum (SFS) that is aggregated across the
populations being compared.
Both of these approaches to summarizing the data result in information about
the model being lost
(i.e., the summary statistics used for inference are insufficient).
% By using the counts of allele patterns from each comparison, our approach is
% able to use all of the information in the data about the model.
By using the mathematical work of \citet{Bryant2012}, our method is able to
calculate the likelihood of the population histories of the comparisons
directly from the counts of character patterns from each population, while
integrating over all possible gene trees under a coalescent model and all
possible mutational histories along those gene trees under a finite-sites model
of character evolution.
Not only does this allow our approach to leverage all of the information in the
data, but it does so efficiently;
when analyzing four divergence comparisons,
\citet{Oaks2018ecoevolity} found this approach requires approximately 9340 times
less computing time than using ABC.
Also, calculating the likelihood of the model from each character pattern
naturally accommodates missing data \citep{Oaks2018ecoevolity}.
In contrast, there is no straightforward way of accounting for missing data
when summarizing genetic data into SFS and other population genetic summary
statistics \citep{Hahn2018}.

% \citet{Chan2014}:
% \begin{itemize}
%     \item Assumes only population expansions
%     \item Assumes at most one simultaneous event (taxa are either part of the
%         event or expand independently)
%     \item Implemenation for single locus data
% \end{itemize}

% \citet{Gehara2017}:
% Same as \citet{Chan2014} but play around with various ways of modelling
% overdispersed timing of events.

% \citet{Xue2015}:
% \begin{itemize}
%     \item Implementation for genomic data (uses SFS as summary statistic)
%     \item Assumes only pop expansions
%     \item Assumes at most one simult event (when applied to sticklebacks)
% \end{itemize}

% \citet{Xue2017}
% \begin{itemize}
%     \item Implementation for genomic data (uses SFS as summary statistic)
%     \item Allows expansion and contraction
%     \item General in regard to the number of shared events
%     \item Time buffering
% \end{itemize}

% \citet{Prates2016}:
% \begin{itemize}
%     \item Populations that expand/contract are prespecified? No clear.
%     \item Assumes at most one simultaneous event (taxa are either part of the
%         event or expand independently)
% \end{itemize}

\section{Conclusions}

% We are threading a needle.
There is a narrow temporal window within which we can reasonably estimate the
time of a demographic change.
The width of this window is determined by how deep in the past the change
occurred relative to the effective size of the population (i.e., in coalescent
units).
If too old or recent, there are too few coalescence events before or after the
demographic change, respectively, to provide information about the effective
size of the population.
When we are careful to simulate data within this window, and the change in
population size is large enough, we can estimate the time of the demographic
changes reasonably well
(e.g., see the top row of \fig{}~\ref{fig:valsimsetimesopt}).
However, even under these favorable conditions, the ability to correctly infer
the shared timing of demographic events among populations is quite limited
(\fig{}~\ref{fig:valsimsmodelopt}).
When only variable characters are analyzed (i.e., SNPs), estimates of the
timing and sharing of demographic changes are consistently bad; we see this
across all the conditions we simulated.
Most alarmingly, when the priors are more diffuse than the distributions that
generated the data, as will be true in most empirical applications, there is a
strong bias toward estimating too few demographic events
(i.e., over-clustering comparisons to demographic events;
Row 2 of \fig{}~\ref{fig:valsimsmodeldiffuse}),
especially when only variable characters are analyzed.
These results help explain the stark contrast we see in our results from the
stickleback RADseq data when including versus excluding constant sites
(\fig{}~\ref{fig:sticklesummary}).
These findings are in sharp contrast to estimating shared \emph{divergence}
times, which is much more accurate, precise, and robust to prior assumptions
\citep[\figs \labelcref{fig:valsimsmodeldiffuse,fig:valsimsetimesdiffuse,fig:mixsims};][]{Oaks2018ecoevolity,Oaks2018paic}.

Given the poor estimates of co-demographic changes, even when all the
information in the data are leveraged by a full-likelihood method, any
inference of shared demographic changes should be treated with caution.
However, there are potential ways that estimates of shared demographic
events could be improved.
For example, as discussed by \citet{Myers2008}, modelling loci of contiguous,
linked sites could help extract more information about past demographic
changes.
Longer loci can contain much more information about the lengths of branches in
the gene tree, which are critically informative about the size of the
population through time.
This is evidenced by the extensive literature on powerful 
``skyline plot'' and ``phylodynamic'' methods
\citep{Pybus2000,Strimmer2001,OpgenRhein2005,Drummond2005,Heled2008,Minin2008beast,Ho2011,Palacios2012,Palacios2012UAI,Stadler2013,Gill2013,Palacios2014,Lan2015,Karcher2016,Karcher2017,Faulkner2018,Karcher2019}.
Obviously, the length of loci will be constrained by recombination.
Nonetheless, with loci from across the genome, each with more information about
the gene tree they evolved along \citep{Speidel2019},
perhaps more information can be captured about temporally clustered changes in
the rate of coalescence across populations.

Another potential source of information could be captured by modelling
recombination along large regions of chromosomes.
By approximating the full coalescent process, many methods have been developed
to model recombination in a computationally feasible manner
\citep{McVean2005,Marjoram2006,Chen2009,Li2011,Sheehan2013,Schiffels2014,Rasmussen2014,Palacios2015}.
This could potentially leverage additional information from genomic data about the
linkage patterns among sites along chromosomes.

The inference of shared evolutionary events could also stand to benefit from
information about
past environmental conditions,
life history data about the taxa,
and ecological data about how they interact.
Modeling ecological aspects of the taxa and historical environmental conditions
could provide important information about which comparisons are most likely to
respond to environmental changes and when, and which taxa are likely to
interact and influence each other's demographic trajectories.
While collecting these types of data and modelling these sorts of dynamics
is challenging, approximate approaches can help to lead the way
\citep{He2013,Massatti2016,Bemmels2016,Knowles2017,Papadopoulou2016}.
All of the these avenues are worth pursuing given the myriad historical
processes that predict patterns of temporally clustered demographic changes
across species.
