% During an ``event'' at time \etime, one or more demographic changes and/or
% divergences can occur.
% We estimate the number and timing of events and the assignment of comparisons
% to those events under a Dirichlet-process \citep{Ferguson1973,Antoniak1974}
% prior model.
% The \emph{a priori} tendency for comparisons to share events is controlled by
% the concentration parameter (\concentration) of the Dirichlet process.
% We use Markov chain Monte Carlo (MCMC) algorithms
% \citep{Metropolis1953,Hastings1970,Neal2000}
% to sample from the joint posterior of the model.
% See Appendix~\ref{appendix:model} for a full description of the model, and
% Table~\ref{table:notation} for a key to the notation we use throughout this
% paper.

\subsection{Analyses of simulated data}

\subsubsection{Assessing ability to estimate timing and sharing of demographic changes}

We used the \simcoevolity and \ecoevolity tools within the \ecoevolity software
package
\citep{Oaks2018ecoevolity}
to simulate and analyze \datasets, respectively, under a variety of conditions.
Each simulated \dataset comprised 500,000 unlinked biallelic characters from 10
diploid individuals (20 genomes) sampled per population from three demographic
comparisons.
We specified the concentration parameter of the Dirichlet process so that
the mean number of events was 2 ($\concentration = 1.414216$).
We assumed the mutation rates of all three populations were equal and 1, such
that time and effective population sizes were scaled by the mutation rate.
When analyzing each simulated \dataset, we ran four MCMC chains for 75,000
generations with a sample taken every 50 generations.
From preliminary analyses, we calculated the potential scale reduction factor
\citep[PSRF; the square root of Equation 1.1 in][]{Brooks1998} and effective
sample size \citep{Gong2014} from the four chaings for all continuous
parameters and the log likelihood using the \texttt{pyco-sumchains} tool of
\pycoevolity (Version 0.1.2 Commit 89d90a1).
Based on these metrics of MCMC convergence and mixing, we summarized the last
1000 samples from each chain for a total of 4000 samples of parameter values to
approximate the posterior distribution for ever simulated data set.

%% Initial simulation conditions removed

% \paragraph{Simulation conditions chosen to improve performance}
Initially, we simulated data under conditions we thought spanned parameter
space that would be conducive and challenging for estimating the timing and
sharing of demograpic changes, but estimates were quite poor across all the
initial simulation conditions (see Supporting Information).
In an effort to find conditions under which the timing and sharing of
demographic changes could be better estimated, and avoid combinations of
parameter values that caused identifiability problems in our initial
analyses,
we next explored simulations under distributions on times and population sizes
offset from zero, and with much more recent demographic event times.
% Event time ~ gamma(shape=4.0, scale=0.000475, offset=0.0001); mean 0.002
% relative root size ~ gamma(shape=5.0, scale = 0.04, offset = 0.05); mean 0.25
% relative root size ~ gamma(shape=5.0, scale = 0.09, offset = 0.05); mean 0.5
% relative root size ~ gamma(shape=5.0, scale = 0.79, offset = 0.05); mean 4
% relative root size ~ gamma(shape=5.0, scale = 0.19, offset = 0.05); mean 1
% relative root size ~ gamma(shape=50.0, scale = 0.02, offset = 0.0); mean 1
For the mutation-scaled effective size of the descendant
population
($\epopsize[\descendantpopindex{}]\murate$),
we used an offset gamma distribution with a shape of 4, offset of 0.0001, and
mean of 0.0021 (accounting for the offset).
For the distribution of event times, we used a gamma distribution
with a shape of 4, offset of 0.0001, and a mean of 0.002 (accounting
for the offset; 0.25 units of $4N_e$ generations, on average).
Again, we used five different distributions on the relative effective size of
the ancestral population (see left column of
\figs
\labelcref{fig:valsimsmodelopt,fig:valsimsetimesopt}):
\begin{enumerate}[label=B.\arabic*]
    \item \dogamma{5}{0.25}{0.05} (4-fold population increase) \label{sims:optimalFourFoldIncrease}
    \item \dogamma{5}{0.5}{0.05} (2-fold population increase)  \label{sims:optimalTwoFoldIncrease}
    \item \dogamma{5}{4}{0.05} (4-fold population decrease)    \label{sims:optimalFourFoldDecrease}
    \item \dogamma{5}{1}{0.05} (no change on average, but a fair amount of variation) \label{sims:optimalCenter}
    \item \dogamma{50}{1}{0} (no change on average, little variance) \label{sims:optimalCenterNarrow}
\end{enumerate}
We generated 500 \datasets under each of these five distributions, and analyzed
all of them using priors that matched the generating distributions.

\subsubsection{Simulations to assess sensitivity to prior assumptions}

% Sim disributions:
% Event time ~ gamma(shape=4.0, scale=0.000475, offset=0.0001); mean 0.002
% relative root size ~ gamma(shape=5.0, scale = 0.04, offset = 0.05); mean 0.25
% descendant size ~ gamma(4.0, scale=0.0005, offset = 0.0001); mean 0.0021
% Priors:
% Event time ~ exponential(mean = 0.005)
% relative root size ~ Exponential(mean = 2.0)
% descendant size ~ gamma(2.0, scale 0.001); mean 0.002

Next, we simulated an additional 500 \datasets under
Condition~\ref{sims:optimalFourFoldIncrease} above.
We then analyzed each of these \datasets under ``diffuse'' prior
distributions:
\begin{itemize}
    \item $\etime \sim \dexponential{0.005}$
    \item $\rootrelativepopsize \sim \dexponential{2}$
    \item $\epopsize[\descendantpopindex{}] \sim \dgamma{2}{0.002}$
\end{itemize}
These distributions were chosen to reflect realistic amounts of prior
uncertainty about the timing of demographic changes and past and present
effective population sizes when analyzing empirical data.
For comparison, we also performed the same simulations and analyses under
diffuse priors for three divergence comparisons.
For these divergence comparisons, we simulated 10 sampled genomes per
population to match the same total number of samples per comparison (20) as the
demographic simulations.


\subsubsection{Simulating a mix of divergence and demographic comparisons}

To explore how well our method can infer a mix of shared demographic changes
and divergence times, we simulated 500 \datasets comprised of 6 comparisons:
3 demographic comparisons and
3 divergence comparisons.
To ensure the same amount of data across comparisons, we simulated
20 sampled genomes (10 diploid individuals) from each comparison
(i.e., 10 genomes from both populations of each divergence comparison).
We used the same simulation conditions described above for
\ref{sims:optimalTwoFoldIncrease},
and specified these same distributions as priors when analyzing all of the
simulated \datasets.


\subsubsection{Simulating linked sites}
To assess the effect of linked sites on the inference
of the timing and sharing of demographic changes,
we simulated \datasets comprising 5000 100-base-pair
loci (500,000 total characters).
The distributions on parameters were the same
as the conditions described for \ref{sims:optimalFourFoldIncrease} above.
These same distributions were used as priors when
analyzing the simulated \datasets.

% \subsection{Data-acquisition bias?}


\subsection{Empirical application to stickleback data}


\subsubsection{Assembly of loci}
We assembled the publicly available RADseq data collected by
\citet{Hohenlohe2010}
from five populations of threespine sticklebacks (\spp{Gasterosteus aculeatus})
from south-central Alaska.
After downloading the reads mapped to the stickleback genome by
\citet{Hohenlohe2010}
from Dryad
(doi:10.5061/dryad.b6vh6),
We assembled reference guided alignments of loci in Stacks v1.48
\citet{Catchen2013} with a minimum read depth of 3 identical reads per locus
within each individual and the bounded single-nucleotide polymorphism (SNP)
model with error bounds between
0.001 and 0.01.
To maximize the number of loci and minimize paralogy, we assembled each
population separately;
because \ecoevolity models each population separately
(\fig{}~\ref{fig:modelCartoon}),
the characters do not need to be orthologous across populations, only within
them.

\subsubsection{Inferring shared demographic changes with \ecoevolity}

We used a value for the concentration parameter of the Dirichlet process
that corresponds to a mean number of three events
($\concentration = 2.22543$).
We used the following prior distributions on the timing of events and effective
sizes of populations:
\begin{itemize}
    \item $\etime \sim \dexponential{0.001}$
    \item $\rootrelativepopsize \sim \dexponential{1}$
    \item $\epopsize[\descendantpopindex{}] \sim \dgamma{2}{0.002}$
\end{itemize}
To assess the sensitivity of the results to these prior assumptions,
we also analyzed the data under two additional priors on
the concentration parameter, event times, and relative
effective population size of the ancestral population:
\begin{itemize}
    \item $\concentration = 13$ (half of prior probability on 5 events)
    \item $\concentration = 0.3725$ (half of prior probability on 1 event)
    \item $\etime \sim \dexponential{0.0005}$
    \item $\etime \sim \dexponential{0.01}$
    \item $\rootrelativepopsize \sim \dexponential{0.5}$
    \item $\rootrelativepopsize \sim \dexponential{0.1}$
\end{itemize}

For each prior setting, we ran 10 MCMC chains for 150,000 generations, sampling
every 100 generations; we did this using all the sites in the assembled
stickleback loci and only SNPs.
To assess convergence and mixing of the chains, we calculated the potential
scale reduction factor \citep[PSRF; the square root of Equation 1.1 in][]{Brooks1998}
and effective sample size \citep{Gong2014} of all continuous parameters and the
log likelihood using the \texttt{pyco-sumchains} tool of \pycoevolity
(Version 0.1.2 Commit 89d90a1).
We also visually inspected the sampled log likelihood and parameters values
over generations with the program Tracer Version 1.6 \citep{Tracer16}.
The MCMC chains for all analyses converged almost immediatley; we
conservatively removed the first 101 samples from each chain, resulting in
14,000 samples from the posterior (1400 samples from 10 chains) for each
analysis.
