\subsection{Analyses of simulated data}

\subsubsection{Estimating timing and sharing of demographic changes}

% Under our initial simulation conditions,
% that we
% selected to comprise a mix of conditions both favorable and challenging
% for estimating the timing and sharing of demographic changes
Despite our attempt to capture a mix of favorable and challenging parameter
values in our initial simulation condtions (Table~S\ref{table:prelimsimconditions}),
estimates of the timing and sharing of demographic events were quite poor
across all the simulation conditions we initially explored
(\figs{}~S\labelcref{fig:valsimsetimesinitial,fig:valsimsmodelinitial,fig:valsimsasizesinitial,fig:valsimsdsizesinitial}).
Even after we tried selecting simulation conditions that are more favorable for
identifying the event times, estimates of the timing and
sharing of demographic events remain quite poor
(\figs \ref{fig:valsimsetimesopt} and \ref{fig:valsimsmodelopt}).
Under the recent (but not too recent) 4-fold population-size increase (on
average) scenario, we do see better estimates of the times of the demographic
change
(\vsimfourinc; top row of \fig{}~\ref{fig:valsimsetimesopt}),
but the ability to identify the correct number of events and the assignment of
the populations to those events remains quite poor;
the correct model is preferred only 57\% of the time, and the
median posterior probability of the correct model is only 0.42
(top row of \fig{}~\ref{fig:valsimsmodelopt}).
Under the most extreme population retraction scenario
(\vsimfourdec; 4-fold, on average),
the correct model is preferred only 40\% of the time, and the median
posterior probability of the correct model is only 0.26
(middle row of \fig{}~\ref{fig:valsimsmodelopt}).
Estimates are especially poor when using only variable characters,
so we focus on the results using all characters
(second versus third column of \figs
\ref{fig:valsimsetimesopt}
and
\ref{fig:valsimsmodelopt});
we also see worse estimates of population sizes when excluding invariant
characters
(\figs
S\ref{fig:valsimsasizesopt}
and
S\ref{fig:valsimsdsizesopt}).

\ifembed{
\input{fig-sims-event-times-opt.tex}
}{}

\ifembed{
\input{fig-sims-model-opt.tex}
}{}

Under the ``worst-case'' scenario of little population-size change
(\vsimnochange; bottom row of \figs \ref{fig:valsimsetimesopt} and
\ref{fig:valsimsmodelopt}),
our method is unable to identify the timing or model of demographic change.
As expected, under these conditions our method returns the prior on the timing
of events (bottom row of \fig{}~\ref{fig:valsimsetimesopt})
and always prefers either a model with a single, shared demographic
event (model "000") or independent demographic changes (model "012";
bottom row of \fig{}~\ref{fig:valsimsmodelopt}).
This is expected behavior, because there is essentially no information in the
data about the timing of demographic changes, and a Dirichlet process with a
mean of 2.0 demographic events, puts approximately 0.24 of the prior
probability on the models with one and three events, and 0.515 prior
probability on the three models with two events (approximately 0.17 each).
Thus, with little information, the method samples from the prior distribution
on the timing of events, and prefers one of the two models with larger prior
probability.

Doubling the number of individuals sampled per population to 20 had very little
affect on the results
(\fig{}~S\ref{fig:valsimspopsamplesize}).
Likewise, doubling the number of demographic comparisons to six had no affect
on the accuracy or prescision of estimating the timing of demographic changes
or effective population sizes
(Rows 1, 3, and 4 of \fig{}~S\ref{fig:valsimscompsamplesize}).
The ability to infer the correct number of demographic events,
and assignment of population to the events (\etimesets),
is much worse when there are six comparisons, which is
not surprising given that the number of possible assignments
of populations to events is 203 for six comparisions, compared
to five for three comparisons \citep{Bell1934}.

The 95\% credible intervals of all the parameters
covers the true value approximately 95\% of the time
(\figs
\ref{fig:valsimsetimesopt},
S\ref{fig:valsimsasizesopt},
and
S\ref{fig:valsimsdsizesopt}).
Given that our priors match the underlying distributions that generated the
data, this coverage behavior is expected, and is an important validation
of our implemetation of the model and corresponding MCMC algorighms.


\subsubsection{Sensitivity to prior assumptions}

Above, we observe the best estimates of the timing and sharing of demographic
events under narrow distributions on the relative effective size of the
ancestral population
(top row of \figs
\labelcref{fig:valsimsetimesinitial,fig:valsimsmodelinitial,fig:valsimsetimesopt,fig:valsimsmodelopt}),
which were used to both simulate the data and as the prior
when those data were analyzed.
Thus, the improved behavior could be due to these narrow prior distributions that
are unrealistically informative for empirical studies, for which there is
usually little \emph{a priori} information about past population sizes.
When we analyze data under more realistic, diffuse priors, estimates of shared
demographic events deteriorate, whereas divergence-time estimates remain
robust.
% Our results show that the better performance under these distributions was at
% least partially caused by greater prior information, and that
% inference of shared demographic events is much more sensitive to
% prior assumptions than shared divergences
(\figs \labelcref{fig:valsimsmodeldiffuse,fig:valsimsetimesdiffuse}).
% To determine whether the better performance under these distributions was
% caused by more informative data or priors, we simulated \datasets under a
% narrow distribution on the relative ancestral population size, and analyzed
% these \datasets under more realistic, ``diffuse'' prior distributions on
% populations sizes and event times.
% This will also allow us to assess how sensitive estimates of the timing and
% sharing of demographic events are to prior assumptions.
% For comparison, we repeated the same simulations for pairs of populations for
% which we estimated the timing and sharing of their divergences (the same total
% number of gene copies were sampled from each pair).
% Sim disributions:
% Event time ~ gamma(shape=4.0, scale=0.000475, offset=0.0001); mean 0.002
% relative root size ~ gamma(shape=5.0, scale = 0.04, offset = 0.05); mean 0.25
% descendant size ~ gamma(4.0, scale=0.0005, offset = 0.0001); mean 0.0021
% Priors:
% Event time ~ exponential(mean = 0.005)
% relative root size ~ Exponential(mean = 2.0)
% descendant size ~ gamma(2.0, scale 0.001); mean 0.002
Specifically, the precision of time estimates of demographic changes decreases
substantially under the diffuse priors
(top two rows of \fig{}~\ref{fig:valsimsetimesdiffuse}),
whereas the precision of the divergence-time estimates
is high and largely unchanged under the diffuse priors
(bottom two rows of \fig{}~\ref{fig:valsimsetimesdiffuse}).
We see the same patterns in the estimates of population sizes
(\figs
S\ref{fig:valsimsasizesdiffuse}
and
S\ref{fig:valsimsdsizesdiffuse}).

\ifembed{
\input{fig-sims-event-times-diffuse.tex}
}{}

Furthermore, under the diffuse priors, the probability of inferring the correct
model of demographic events decreases from 0.57 to 0.434 when all characters
are used, and from 0.36 to 0.284 when only variable characters are used
(top two rows of \fig{}~\ref{fig:valsimsmodeldiffuse}).
The median posterior probability of the correct model also decreases from
0.422 to 0.292 when all characters are used,
and from 0.231 to 0.178 when only variable characters are used
(top two rows of \fig{}~\ref{fig:valsimsmodeldiffuse}).
\emph{Most importantly, we see a strong bias toward underestimating the number
    of events under the more realistic diffuse priors}
(top two rows of \fig{}~\ref{fig:valsimsmodeldiffuse}).
In comparison, the inference of shared divergence times is much more accurate,
precise, and robust to the diffuse priors
(bottom two rows of \fig{}~\ref{fig:valsimsmodeldiffuse}).
When all characters are used, under both the correct and diffuse
priors, the correct divergence model is preferred over 91\% of the time,
and the median posterior probability of the correct model is over
0.93.

\ifembed{
\input{fig-sims-model-diffuse.tex}
}{}


\subsection{Inferring a mix of shared divergences and demographic changes}

When demographic and divergence comparisons are analyzed separately, the
performance of estimating the timing and sharing of demographic changes and
divergences is dramatically different, with the latter being much more accurate
and precise than the former
(e.g., see
\figs
\ref{fig:valsimsetimesdiffuse}
and
\ref{fig:valsimsmodeldiffuse}).
One might hope that
if we analyze a mix of demographic and divergence comparisons,
the informativeness of the divergence times can help ``anchor'' and
improve the estimates of shared demographic changes.
However, our results from simulating \datasets comprising a mix of three
demographic and three divergence comparisons rule out this possibility.
% To explore this possibility, we simulated datasets with six comparisons,
% comprising a mix of three populations that experienced a demographic change and
% three pairs of populations that diverged.
% For these mixed-comparison simulations, we used a gamma distribution on event
% times with a shape of 4, offset of 0.0001, and a mean of 0.002 substitutions per
% site (accounting for the offset; 0.25 units of $4N_e$ generations on average).
% For the distribution on the relative size of the ancestral population,
% we used a gamma distribution with a shape of 5, an offset of 0.05, and a mean
% of 0.5; a 2-fold population size increase on average.
% These are the same distributions used for the second row of
% \figs
% \ref{fig:valsimsetimesopt}
% and
% \ref{fig:valsimsmodelopt}.

% We summarized the timing and sharing of the demographic changes
% separately from the divergences so that we could determine whether the
% divergence-time estimates could help improve the estimates of the
% times of the demographic changes.
When analyzing a mix of demographic and divergence comparisons, the ability to
infer the timing and sharing of demographic changes remains poor, whereas
estimates of shared divergences remain accurate and precise
(\fig{}~\ref{fig:mixsims}).
The estimates of the timing and sharing of demographic events are nearly
identical to when we simulated and analyzed only three demographic comparisons
under the same distributions on event times and population sizes
(compare left column of \fig{}~\ref{fig:mixsims}
to the second row of \figs
\ref{fig:valsimsetimesopt}
and
\ref{fig:valsimsmodelopt}).
The same is true for the estimates of population sizes
(\fig{}~S\ref{fig:mixsimsfull}).
Thus, there does not appear to be any mechanism by which the more informative
divergence-time estimates ``rescue'' the estimates of the timing and sharing of
the demographic changes.

\ifembed{
\input{fig-sims-mix.tex}
}{}


\subsection{The effect of linked sites}

Most reduced-representation genomic datasets are comprised of loci of
contiguous, linked nucleotides.
Thus, when using the method presented here that assumes each character is
effectively unlinked,
% (i.e., evolved along a gene tree that is independent from other characters,
% conditional on the population history)
one either has to violate this assumption, or discard all but (at most) one
site per locus.
Given that all the results above indicate better estimates when all
characters are used (compared to using only variable characters), we
simulated linked sites to determine which strategy is better:
analyzing all linked sites and violating the assumption of unlinked characters,
or discarding all but (at most) one variable character per locus.

% To do this, we repeated the most favorable simulation conditions (on average
% 4-fold population expansion; see first row of
% \figs
% \ref{fig:valsimsetimesopt}
% and
% \ref{fig:valsimsmodelopt}),
% except that 100 characters were simulated along 5000
% simulated gene trees.
% In other words, the simulated \datasets comprised 5000
% 100-base-pair loci, rather than 500,000 unlinked sites.
The results are almost identical to when all the sites were unlinked
(compare first row of
\figs
\ref{fig:valsimsetimesopt}
and
\ref{fig:valsimsmodelopt}
to
\fig{}~\ref{fig:locisims},
and the first row of
\figs
S\ref{fig:valsimsasizesopt}
and
S\ref{fig:valsimsdsizesopt}
to the bottom two rows of
\fig{}~S\ref{fig:locisimsfull}).
Thus, violating the assumption of unlinked sites has little
affect on the estimation of the timing and sharing of
demographic changes;
this is also true for estimates of population sizes
(\fig{}~S\ref{fig:locisimsfull}).
This is consistent with the findings of
\citet{Oaks2018ecoevolity} and
\citet{Oaks2018paic}
that linked sites had little affect on the estimation of
shared divergence times.
These results suggest that analyzing all of the sites in loci assembled from
reduced-representation genomic libraries (e.g., sequence-capture or RADseq
loci) is a better strategy than excluding sites to avoid violating the
assumption of unlinked characters.

\ifembed{
\input{fig-sims-loci.tex}
}{}


% \subsection{Data-acquisition bias?}


\subsection{Reassessing the co-expansion of stickleback populations}

Using an ABC analog to the model of shared demographic changes developed here,
\citet{Xue2015} found very strong support (0.99 posterior probability) that
five populations of threespine sticklebacks (\spp{Gasterosteus aculeatus})
from south-central Alaska recently
co-expanded.
This inference was based on the publicly available RADseq data collected by
\citet{Hohenlohe2010}.
% (\url{https://trace.ncbi.nlm.nih.gov/Traces/sra/?study=SRP001747};
% NCBI Short Read Archive accession numbers SRX015871--SRX015877).
We re-assembled and analyzed these data under our full-likelihood
Bayesian framework, both using all sites from assembled loci
and only SNPs.


Stacks produced a concatenated alignment with
2,115,588,
2,166,215,
2,081,863,
2,059,650, and
2,237,438
total sites, of which
118,462,
89,968,
97,557,
139,058, and
103,271
were variable for the Bear Paw Lake, Boot Lake, Mud Lake, Rabbit Slough, and
Resurrection Bay stickleback populations respectively.
When analyzing all sites from the assembled stickleback
RADseq data, we find strong support for five independent
population expansions (no shared demographic events;
\fig{}~\ref{fig:sticklesummary}).
In sharp contrast, when analyzing only SNPs, we find
support for a single, shared, extremely recent population expansion
(\fig{}~\ref{fig:sticklesummary}).
The support for a single, shared event is consistent with the results from our
simulations using diffuse priors and only including SNPs, which showed
consistent, spurious support for a single event
(Row 2 of \fig{}~\ref{fig:valsimsmodeldiffuse}).
These results are relatively robust to a broad range of prior
assumptions
(\figs
S\labelcref{%
fig:sticklebydppevents,fig:sticklebydpptimes,fig:sticklebydppsizes,%
fig:sticklebytimeevents,fig:sticklebytimetimes,fig:sticklebytimesizes,%
fig:sticklebysizeevents,fig:sticklebysizetimes,fig:sticklebysizesizes,%
}).

\ifembed{
\input{fig-sticklebacks.tex}
}{}

When using only SNPs, estimates of the timing of the single, shared demographic
event are essentially at the minimum of zero, suggesting that there is little
information about the timing of any demographic changes in the SNP data alone.
This is consistent with results of \citet{Xue2015} where the single, shared
event was also estimated to have occurred at the minimum (1000 generations) of
their uniform prior on the timing of demographic changes.
In light of our simulation results, the support for a single event based solely
on SNPs, seen here and in \citet{Xue2015}, is likely caused by a combination of
(1) misspecified priors, and
(2) the lack of information about demographic history when invariant characters
are discarded.
Our estimates using all of the sites in the stickleback RADseq loci should be
the most accurate, according to our results from simulated data.
However, the unifying theme from our simulations is that all estimates of
shared demographic events tend to be poor and should be treated with a lot of
skepticism.
% Given our results when using all the information from data simulated under
% conditions favorable for estimating the timing of demographic changes, strong
% posterior support for any particular scenario is almost certainly spurious.
